driver-core/src/test/unit/com/mongodb/ConnectionStringSpecification.groovy

```diff
@@ -131,6 +124,35 @@ class ConnectionStringSpecification extends Specification {
         type << ['amp', 'semi', 'mixed']
     }
 
+    @Unroll
```

Comment:
These are the explicit failure cases.  5 might have succeeded on master but then they would have later failed when opening - eg unix socket support, malformed hosts, invalid ports, multiple at signs or colons.